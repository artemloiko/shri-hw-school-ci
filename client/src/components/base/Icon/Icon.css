.icon {
  min-width: 16px;
  min-height: 16px;
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
  background-image: url("data:image/svg+xml,%3Csvg version='1.0' xmlns='http://www.w3.org/2000/svg' width='16' height='16' viewBox='0 0 512 512'%3E%3Cpath d='M64.8 1c-17 2.9-29.2 9.3-41.8 22C12.2 33.7 6.8 42.7 2.8 56.6L.5 64.5v383l2.3 7.9c4 13.9 9.4 22.9 20.2 33.6 10.7 10.8 19.7 16.2 33.6 20.2l7.9 2.3h383l7.9-2.3c13.9-4 22.9-9.4 33.6-20.2 10.8-10.7 16.2-19.7 20.2-33.6l2.3-7.9v-383l-2.3-7.9c-4-13.9-9.4-22.9-20.2-33.6-10.5-10.6-19.3-15.9-33-20.2L448.5.5 259 .4C154.8.3 67.4.6 64.8 1zm379.6 52.1c3.8 1.4 6.8 3.4 9.5 6.5 7.5 8.5 7.1 2.2 7.1 112.8v99l-11.2-11c-6.2-6.1-14-12.7-17.3-14.7-18-10.7-42-12.7-61.1-5-11.7 4.6-18.2 9.8-42.9 34.4L304.4 299l-34-32.8c-49.8-47.9-50.4-48.4-58-52.4-12.1-6.3-20.2-8.3-33.9-8.2-10.1.1-13.3.5-20 2.7-16 5.3-14.2 3.8-78.5 64L51.5 299l-.3-112.7C51 58.6 50.4 67.5 58.9 59s-9.1-7.8 196.8-7.9c182.6-.1 183.2-.1 188.7 2zM186.5 257.9c2.7 1.2 38.6 35.3 107 101.6 56.7 54.9 103.2 100.2 103.3 100.6.2.5-70.9.9-161.5.9-180.7 0-167 .6-175.7-7.1-3.1-2.7-5.1-5.7-6.5-9.5-2-5.1-2.1-7.6-2.1-40.2l.1-34.7 49.5-46.5c67.9-63.7 69.1-64.9 72.7-65.9 5-1.4 8.9-1.2 13.2.8zm217.3 30.6c2.4 1.1 12.8 10.6 30.5 28.3l26.7 26.7v46.8c0 44.5-.5 54.7-3 56.2-.6.4-70.7-66.7-112.5-107.6l-4-3.9 22-22.1c13.8-13.9 23.5-22.8 26-24 4.9-2.3 9.4-2.4 14.3-.4z'/%3E%3Cpath d='M308.8 129.9c-3.1 1-7.2 2.8-9.1 4-5.2 3.2-11.6 10.6-14.5 16.9-2.4 5-2.7 7-2.7 15.7 0 9.3.2 10.5 3.3 16.7 3.7 7.5 10.5 14.3 17.8 17.8 18.9 8.8 41.4 1 50.6-17.8 3.1-6.2 3.3-7.4 3.3-16.7 0-9.3-.2-10.5-3.3-16.7-8.3-16.9-28.2-25.6-45.4-19.9zm19.7 27.4c1.9 1.9 3.7 4.7 4.1 6.2 1.1 4.3-1.5 10.5-5.5 13.2-7 4.7-16.3 1.6-19-6.3-1.8-5.2-.9-8.8 3.4-13.1 2.9-2.9 3.9-3.3 8.5-3.3s5.6.4 8.5 3.3z'/%3E%3C/svg%3E");
}

.icon_size_small {
  min-width: 12px;
  min-height: 12px;
}

.icon_type_commit {
  background-image: url('../../../images/icon/code-commit.svg');
}

.icon_type_calendar {
  background-image: url('../../../images/icon/calendar.svg');
}

.icon_type_stopwatch {
  background-image: url('../../../images/icon/stopwatch.svg');
}

.icon_type_user {
  background-image: url('../../../images/icon/user.svg');
}

.icon_type_rebuild {
  background-image: url('../../../images/icon/rebuild.svg');
}

.icon_type_play {
  background-image: url('../../../images/icon/play.svg');
}

.icon_type_settings {
  background-image: url('../../../images/icon/settings.svg');
}
