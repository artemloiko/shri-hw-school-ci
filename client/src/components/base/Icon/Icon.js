import React from 'react';
import PropTypes from 'prop-types';
import { cn } from 'utils/bem-cn';

import './Icon.css';

function Icon(props) {
  return <div className={cn('icon', props)}></div>;
}

Icon.propTypes = {
  className: PropTypes.string,
  mods: PropTypes.shape({
    size: PropTypes.oneOf(['small']),
    type: PropTypes.oneOf([
      'commit',
      'calendar',
      'stopwatch',
      'user',
      'rebuild',
      'play',
      'settings',
    ]).isRequired,
  }).isRequired,
  mix: PropTypes.arrayOf(PropTypes.string),
};

export default Icon;
