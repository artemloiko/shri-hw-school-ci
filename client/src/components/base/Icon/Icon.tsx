import React from 'react';
import { cn, CNProps } from 'utils/bem-cn';

import './Icon.css';

export type IconTypes =
  | 'commit'
  | 'calendar'
  | 'stopwatch'
  | 'user'
  | 'rebuild'
  | 'play'
  | 'settings';

type IconModsType = {
  size?: 'small';
  type: IconTypes;
};

export type IconProps = CNProps<IconModsType>;

const Icon: React.FC<IconProps> = (props) => {
  return <div className={cn('icon', props)}></div>;
};

export default Icon;
