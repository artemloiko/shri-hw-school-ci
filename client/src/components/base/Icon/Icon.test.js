import React from 'react';
import renderer from 'react-test-renderer';

import Icon from './Icon';

test('Icon with required props matches the snapshot', () => {
  const tree = renderer.create(<Icon mods={{ type: 'calendar' }} />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Icon with all possible props matches the snapshot', () => {
  const tree = renderer
    .create(
      <Icon
        mods={{
          type: 'calendar',
          size: 'small',
        }}
        className="block"
        mix={['block']}
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
