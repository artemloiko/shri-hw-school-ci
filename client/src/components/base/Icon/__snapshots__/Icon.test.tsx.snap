// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon with all possible props matches the snapshot 1`] = `
<div
  className="block icon icon_type_calendar icon_size_small block__icon"
/>
`;

exports[`Icon with required props matches the snapshot 1`] = `
<div
  className="icon icon_type_calendar"
/>
`;
